__integrations__/__tests__/system/cosmos/queryenrichment.test.ts

```diff
@@ -0,0 +1,173 @@
+          );
+        },
+        O.fold(
+          () => {
+            throw new Error("it should be some");
+          },
+          (result) =>
+            expect(result).toMatchObject({
+              id: COLLECTION_ID_KEY,
+              key: COLLECTION_PARTITION_KEY,
+            }),
+        ),
+      ),
+    )();
+  });
+
+  it("should return None when the item is not found - find by id and partition key", async () => {
+    await pipe(
+      createCosmosQueryEnrichmentService(COSMOSDB_CONNECTION_STRING),
+      TE.fromEither,
+      TE.chain((service) =>
+        service.findByKey(
+          COSMOSDB_NAME,
+          COSMOS_COLLECTION_NAME,
+          COLLECTION_ID,
+          COLLECTION_PARTITION_KEY,
+        ),
+      ),
+      TE.bimap(
+        (err) => {
+          new Error(
+            `it should not fail while finding an existing document - ${err.message}`,
+          );
+        },
+        O.fold(
+          () => constVoid(),
+          () => {
+            throw new Error("it should be none");
+          },
+        ),
+      ),
+    )();
+  });
+});
+
+describe("findByLastVersionKey", () => {
+  it("should return Some(unknown) when the item is found with the last version", async () => {
+    return await pipe(
+      createCosmosQueryEnrichmentService(COSMOSDB_CONNECTION_STRING),
+      TE.fromEither,
+      TE.chain((service) =>
+        service.findLastVersionByKey(
+          COSMOSDB_NAME,
+          COSMOS_COLLECTION_NAME,
+          VERSION_FIELD_NAME,
+          VERSION_FIELD_VALUE,
+          COLLECTION_ID_VERSION,
+          COLLECTION_PARTITION_KEY,
+          COLLECTION_PARTITION_KEY_VALUE,
+        ),
+      ),
+      TE.bimap(
+        (err) => {
+          new Error(
+            `it should not fail while finding an existing document - ${err.message}`,
+          );
+        },
+        O.fold(
+          () => {
+            throw new Error("it should be some");
+          },
+          (result) => {
+            expect(result).toMatchObject({
+              id: COLLECTION_ID_VERSION,
+              key: COLLECTION_PARTITION_KEY_VALUE,
+              version: VERSION_FIELD_VALUE,
+            });
+          },
+        ),
+      ),
+    )();
+  });
+
+  it("should return None(unknown) when the item is not found with the specific version", async () => {
+    await pipe(
+      createCosmosQueryEnrichmentService(COSMOSDB_CONNECTION_STRING),
+      TE.fromEither,
+      TE.chain((service) =>
+        service.findLastVersionByKey(
+          COSMOSDB_NAME,
+          COSMOS_COLLECTION_NAME,
+          VERSION_FIELD_NAME,
+          WRONG_VERSION_FIELD_VALUE,
+          COLLECTION_ID_KEY,
+          COLLECTION_PARTITION_KEY,
+          COLLECTION_PARTITION_KEY_VALUE,
+        ),
+      ),
+      TE.bimap(
+        (err) => {
+          new Error(
+            `it should not fail while finding an existing document - ${err.message}`,
+          );
+        },
+        O.fold(
+          () => constVoid(),
```

Comment:
expect?

Reply:
because it's expecting None. O.Fold executes the first branch only if the result is None. So the check is done implicitly.